app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, ChangeEvent, useEffect, useState, useRef, useCallback } from 'react';
+import T from 'i18n-react';
+import { getRecipeByName, createRecipe } from 'components/RecipeManagement/CreateRecipe/services';
+import { debounce } from 'lodash';
+import { IRecipeFormData } from 'components/RecipeManagement/types';
+import RecipeForm, { CREATE_RECIPE } from 'components/RecipeManagement/RecipeForm';
+import { ICreateRecipeProps } from 'components/RecipeManagement/types';
+
+const PREFIX = 'features.WranglerNewUI.RecipeForm.labels';
+const recipeNameRegEx = /^[a-z\d\s]+$/i;
+
+const noErrorState = {
+  isRecipeNameError: false,
+  recipeNameErrorMessage: '',
+};
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeFormData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [isSaveDisabled, setIsSaveDisabled] = useState<boolean>(true);
+  const [recipeNameErrorData, setRecipeNameErrorData] = useState(noErrorState);
```

Comment:
Updating this line of code with the following will allow you to remove the `useEffect` pointed in the next comment. **Using effects should be your last resort for managing application state; most of the time the intended behaviour can be achieved by implementing the effect logic in handlers** (which is generally more performant as it prevents unnecessary re-renders). Ref: [react docs (official)](https://beta.reactjs.org/learn/keeping-components-pure#where-you-_can_-cause-side-effects)

```js
const [recipeNameErrorData, setRecipeNameErrorDataState] = useState(noErrorState);

setRecipeNameErrorData (recipeNameError) {
  setRecipeNameErrorDataState(recipeNameError);
  handleSaveButtonMode(recipeFormData, recipeNameError);
}
```

You also need to make a minor change to the `handleSaveButtonMode` function.

```js
const handleSaveButtonMode = (formData = recipeFormData, nameErrorData = recipeNameErrorData) => {
    if (
      formData.recipeName === '' ||
      formData.description === '' ||
      formData.recipeName?.trim().length === 0 ||
      formData.description?.trim().length === 0 ||
      nameErrorData.isRecipeNameError
    ) {
      setIsSaveDisabled(true);
    } else {
      setIsSaveDisabled(false);
    }
  };
```

Reply:
removed useEffect and changed implementation by using handler